### PR TITLE
Frontend: Import and use Auth context to verify user login state

### LINKED ISSUE #142: Fix flicker when trying to create a game without logging in
# Description
When clicking on 'Create Game" without logging in, the screen flickers and a toast is displayed. This is because the GameProvider on mounting checks the auth state and returns if not logged in. 

We can fix this flicker by checking it in the homepage itself, before navigating to the game page.
(We should leave the GameContext undisturbed)

### PR DESCRIPTION
**Pull request by Spikoftw**
### **Description**

Fixed the issue causing screen flicker when clicking "Create Game" without being logged in.
Added an authentication check on the homepage before navigating to the game creation page.
Displayed a toast with an alert message if the user is not logged in and attempts to create a game and redire t to login page. 

### **Motivation and Context**

Providing immediate feedback via a toast helps users easily understand why their action failed and redirect user to login. 

### **How to Test**

Make sure you are logged out of the application.
Navigate to the homepage.
Click on the "Start Game" button.
Verify that the message "You need to log in to create a game" appears and that there is no screen flicker and that you are redirected to login page. 
Log in to the application.
Click the "Start Game" button again.
Verify that you are redirected to the game creation page without any issues.

### **Related Issues**

Fixes #142 

### **Checklist**

- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have assigned reviewers to this pull request.

### **Screenshots**

https://github.com/shivansh-bhatnagar18/multiplayer-uno/assets/77557266/c475470d-e5cd-46a6-93cf-cca633534ca9

### CODE REVIEW

File: frontend/src/pages/Home.tsx [hunk 1]
```diff
@@ -44,20 +45,32 @@ function JoinGameModalContent() {
...
             </Button>
         </>
     );
 }
+
 function Home() {
     const modal = useModal();
     const navigate = useNavigate();
+    const { isLoggedIn } = useAuth(); // use Auth Context
+    const { open } = useToast(); // Use hook toast to display messages
```
Comment (kuv2707) [inline frontend/src/pages/Home.tsx:59]: The comments are unnecessary and can be removed

**Review by kuv2707 — COMMENTED** [review]
This seems working.
Please fix the commit history as per the docs and includes screenshots/captures in the PR description.

Also making use of markdown checkboxes makes it easier to read the PR description checklist

File: frontend/src/pages/Home.tsx [hunk 2]
```diff
@@ -44,20 +45,32 @@ function JoinGameModalContent() {
...
 
     const CreateGame = () => {
-        // Logic to create a game
-        console.log('Create Game');
-        navigate('/game?type=create');
+        if (isLoggedIn()) {
+            // Logic to create a game
+            console.log('Create Game');
+            navigate('/game?type=create');
+        } else {
```
Comment (kuv2707) [inline frontend/src/pages/Home.tsx:66]: also navigate to the login page here

**Comment by Spikoftw** [comment]
Hello, thanks for the reply, is it ok now ?

File: frontend/src/pages/Home.tsx [hunk 3]
```diff
@@ -4,9 +4,10 @@ import '../index.css';
 import { useModal } from '../library/modal/ModalContext';
 import { useState } from 'react';
 import { useToast } from '../library/toast/toast-context';
+import { useAuth } from '../contexts/AuthContext'; // Import Auth context
```
Comment (kuv2707) [inline frontend/src/pages/Home.tsx:7]: You missed this comment.
In general we should avoid comments where the intent is self-explanatory.

Also the commit message is good structure wise, but it doesn't need mentions of what is imported etc. That is evident by looking at the code.

The commit message should just provide info on the change introduced.

Comment (Spikoftw) [reply frontend/src/pages/Home.tsx:7]: ok ill make those changes

**Comment by Spikoftw** [comment]
Sorry for this, i'm new to this. I hope this time it's the good one. lol :)

**Comment by Spikoftw** [comment]
Do i need to do something else for this issue ?

**Comment by kuv2707** [comment]
Please fix the pr description to have a proper `fixes #issuenumber` clause.

**Comment by Spikoftw** [comment]
Ive made changes to the description. Is it good now ?

**Comment by kuv2707** [comment]
I dont find the fixes clause in the PR description.
And there's a review comment to be addressed. Other than that it looks fine!

**Comment by Spikoftw** [comment]
I'm sorry but i dont get it. i ve seens the others PRs but i still dont get what you expect from me.
Maybe i'm not watching the right thing, can you show me an example ?
You just want me to put the word "fixes" before #149 ?
Sorry i feel stupid ahah xD

**Comment by kuv2707** [comment]
Its okay, its often the little things that bug us.
You got that right now.
And undo the removal of `<string>` as i had mentioned in a comment

**Comment by Spikoftw** [comment]
Thanks!
The removal of <string> ? I don't see the comment about that

**Review by kuv2707 — COMMENTED** [review]
A minor suggestion. I didn't end the review so the suggestion wasn't visible

File: frontend/src/pages/Home.tsx [hunk 4]
```diff
@@ -4,9 +4,10 @@ import '../index.css';
 import { useModal } from '../library/modal/ModalContext';
 import { useState } from 'react';
 import { useToast } from '../library/toast/toast-context';
+import { useAuth } from '../contexts/AuthContext';
 
 function JoinGameModalContent() {
-    const [gameCode, setGameCode] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (Spikoftw) [reply frontend/src/pages/Home.tsx:9]: Yep, got i back.

**Comment by Spikoftw** [comment]
fixed

**Comment by kuv2707** [comment]
Merged, thanks @Spikoftw

**Pull request merged**